apiserver/common/tools.go

```diff
@@ -274,84 +286,69 @@ func (f *ToolsFinder) findTools(args params.FindToolsParams) (coretools.List, er
 
-	// TODO(juju4) - remove this logic
-	// Older versions of Juju publish agent binary metadata based on series.
-	// The following functions convert the result tools list to the required format.
-
-	// For a given list of tools, return those which match the required
-	// os type. If the tools is for a series, first convert the series into
-	// the corresponding os type before doing the match.
-	// eg if ubuntu tools are requested, and we have bionic tools, say they're ubuntu.
-	wantedOSType := args.OSType
-	compatibleOSTypeTools := func() {
-		added := make(map[version.Binary]bool)
-		var matched coretools.List
-		for _, t := range result {
-			converted := *t
-			// t might be for a series so convert to an os type.
-			if !coreos.IsValidOSTypeName(t.Version.Release) {
-				osTypeName, err := coreseries.GetOSFromSeries(t.Version.Release)
-				if err != nil {
-					continue
-				}
-				converted.Version.Release = strings.ToLower(osTypeName.String())
-			}
-			if converted.Version.Release != wantedOSType {
-				continue
-			}
-			if added[converted.Version] {
+		// t might be for a series so convert to an OS type.
+		if !coreos.IsValidOSTypeName(t.Version.Release) {
+			osTypeName, err := coreseries.GetOSFromSeries(t.Version.Release)
+			if err != nil {
 				continue
 			}
-			matched = append(matched, &converted)
-			added[converted.Version] = true
+			converted.Version.Release = strings.ToLower(osTypeName.String())
 		}
-		result = matched
-	}
-
-	// For a given list of tools, return those which match the required series.
-	// If the tools is for an os type, match on the os type of the series.
-	// The returned tools are converted to the requested series.
-	// eg if bionic tools are requested, we return bionic if we have then
-	// or convert ubuntu tools to say they're bionic.
-	compatibleSeriesTools := func() {
-		osType := coreseries.DefaultOSTypeNameFromSeries(args.Series)
-		added := make(map[version.Binary]bool)
-		var matched coretools.List
-		for _, t := range result {
-			converted := *t
-			if coreos.IsValidOSTypeName(t.Version.Release) {
-				if osType != t.Version.Release {
-					continue
-				}
-				converted.Version.Release = args.Series
-			} else if args.Series != t.Version.Release {
-				continue
-			}
-			if added[converted.Version] {
-				continue
-			}
-			matched = append(matched, &converted)
-			added[converted.Version] = true
+
+		if converted.Version.Release != osType {
+			continue
 		}
-		result = matched
+		if added[converted.Version] {
+			continue
+		}
+
+		matched = append(matched, &converted)
+		added[converted.Version] = true
 	}
 
-	// TODO(juju4) - remove this logic
-	// Set up the final metadata conversion if old tools are requested.
-	if args.Number.Major == 2 && args.Number.Minor <= 8 && (args.OSType != "" || args.Series != "") {
-		if args.OSType != "" {
-			defer compatibleOSTypeTools()
+	return matched
+}
+
+func (f *ToolsFinder) resultForSeriesTools(list coretools.List, series string) coretools.List {
```

Comment:
Is it worth keeping a "//TODO(juju4) here?"

Reply:
I think the comment on the calling block covers it.